rpa-mapper/src/main/java/uk/gov/hmcts/cmc/rpa/mapper/DefenceResponseJsonMapper.java

```diff
@@ -26,17 +34,58 @@ public DefenceResponseJsonMapper(DefendantJsonMapper defendantMapper) {
     }
 
     public JsonObject map(Claim claim) {
-        Response response = claim.getResponse().orElseThrow(IllegalStateException::new);
+        Response response = claim.getResponse().orElseThrow(IllegalArgumentException::new);
         String defendantsEmail = claim.getDefendantEmail();
+
         return new NullAwareJsonObjectBuilder()
             .add("caseNumber", claim.getReferenceNumber())
             .add("responseSubmittedOn", DateFormatter.format(claim.getRespondedAt()))
             .add("defenceResponse", defenceResponse(response))
             .add("defendant", defendantMapper.map(response.getDefendant(), claim.getClaimData().getDefendant(), defendantsEmail))
             .add("mediation", isMediationSelected(response))
+            .add("amountAdmitted", getAmountAdmitted(response))
+            .add("payment", mapPayment(response))
             .build();
     }
 
+    private BigDecimal getAmountAdmitted(Response response) {
+        if (response instanceof PartAdmissionResponse) {
+            return ((PartAdmissionResponse) response).getAmount();
+        }
+        return null;
+    }
+
+    private JsonObject mapPayment(Response response) {
+        JsonObjectBuilder jsonObjectBuilder = new NullAwareJsonObjectBuilder();
+        if (ResponseUtils.isAdmissionResponse(response)) {
+            jsonObjectBuilder = retrievePaymentBuilder(jsonObjectBuilder, response);
+            return jsonObjectBuilder.build();
+        } else {
+            return null;
+        }
+    }
+
+    private JsonObjectBuilder retrievePaymentBuilder(JsonObjectBuilder jsonObjectBuilder, Response response) {
+        PaymentIntention paymentIntention = null;
+        if (response instanceof PartAdmissionResponse) {
+            PartAdmissionResponse partAdmissionResponse = (PartAdmissionResponse) response;
+            paymentIntention = partAdmissionResponse.getPaymentIntention().orElseThrow(IllegalArgumentException::new);
+        } else if (response instanceof FullAdmissionResponse) {
+            FullAdmissionResponse fullAdmissionResponse = (FullAdmissionResponse) response;
+            paymentIntention = fullAdmissionResponse.getPaymentIntention();
+        }
```

Comment:
else NPE? Suggest making the method signature indicate that only admission responses are valid
`retrieveAdmissionsPaymentBuilder(...)` perhaps

Reply:
No NPE because it will be null for non admission responses. Changed method name. Added another NPE check at different place though